src/collections/_documentation/development/sdk-dev/envelopes.md

```diff
@@ -0,0 +1,448 @@
+  considered an arbitrary character not part of the newline, and thus part of
+  the payload.
+- After length-prefixed item payloads there is always a `\n` or EOF. That
+  newline is not considered part of the payload. If there is any other
+  character, including whitespace, the Envelope is malformed.
+
+Item-only example without Envelope headers:
+
+```
+{"type":"attachment","length":10}\n
+helloworld
+```
+
+### Full Examples
+
+These examples contain full Envelope payloads. Newlines are explicitly marked
+with `\n`, unprintable characters are escaped with `\x<><>`. all other
+characters are literal.
+
+**Envelope with 2 items:**
+
+Note that the attachment contains a Windows newline at the end of its
+payload which is included in `length`:
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc","dsn":"https://e12d836b15bb49d7bbf99e64295d995b:@sentry.io/42"}\n
+{"type":"attachment","length":10,"content_type":"text/plain","filename":"hello.txt"}\n
+\xef\xbb\xbfHello\r\n\n
+{"type":"event","length":41,"content_type":"application/json","filename":"application.log"}\n
+{"message":"hello world","level":"error"}\n
+```
+
+**Empty Envelope without items, last newline omitted:**
+
+Note that the attachment contains a Windows newline at the end of its
+payload which is included in `length`:
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc","dsn":"https://e12d836b15bb49d7bbf99e64295d995b:@sentry.io/42"}\n
+{"type":"attachment","length":10,"content_type":"text/plain","filename":"hello.txt"}\n
+\xef\xbb\xbfHello\r\n\n
+{"type":"event","length":41,"content_type":"application/json","filename":"application.log"}\n
+{"message":"hello world","level":"error"}
+```
+
+**Envelope with 2 empty attachments:**
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc"}\n
+{"type":"attachment","length":0}\n
+\n
+{"type":"attachment","length":0}\n
+\n
+```
+
+**Envelope with 2 empty attachments, last newline omitted:**
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc"}\n
+{"type":"attachment","length":0}\n
+\n
+{"type":"attachment","length":0}\n
+```
+
+**Item with implicit length, terminated by newline:**
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc"}\n
+{"type":"attachment"}\n
+helloworld\n
+```
+
+**Item with implicit length, last newline omitted, terminated by EOF:**
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc"}\n
+{"type":"attachment"}\n
+helloworld
+```
+
+**Envelope without headers, implicit length, last newline omitted, terminated by
+EOF:**
+
+```
+{}\n
+{"type":"session"}\n
+{"started": "2020-02-07T14:16:00Z","attrs":{"release":"sentry-test@1.0.0"}}
+```
+
+## Data Model
+
+Each Envelope consists of headers and a potentially empty list of items, each
+with their own headers. Which Headers are required depends on the Items in an
+Envelope. This section describes all Item types and their respective required
+headers.
+
+The type of an item is declared in the `type` header, as well as the payload
```

Comment:
note that we aren't being consistent with Item v item. In general, better to avoid overcapitalization (which is sort-of German, actually, where I believe all Nouns are Capitalized).

For readability, I'd suggest item, not Item. Similarly, header, not Header.

Reply:
Good point, I explicitly uppercased "Envelope" but forgot to check "Item" and "Header". I'd prefer to leave uppercase to avoid confusion with any other meaning of the generic word "item".